### DISCUSSION TITLE
`--assemble-contigs-by` what to specify to run `mixcr findAlleles`

### DISCUSSION THREAD

**Discussion by analeigh-gui** [milaboratory/mixcr#1006] [Q&A] [answered]
I'm following the guide: [Immunoglobulin lineage trees reconstruction ](https://mixcr.com/mixcr/guides/ig-trees-reconstruction/#data)

I have BD Rhapsody full-length BCR mouse data. I first run `mixcr analyze` using the preset.

```
mixcr analyze bd-rhapsody-mouse-tcr-full-length \
        20220923-Mm023B-BCR_R1.fastq.gz \
        20220923-Mm023B-BCR_R2.fastq.gz \
       mouse_vdj/result
```

Then I ran:
```
mixcr findAlleles \
        --report findAlleles.report.txt \
        --json-report findAlleles.report.json \
        --export-alleles-mutations alleles.tsv \
        --export-library alleles.json \
        --output-template reassigned.clns \
        result.contigs.cls
```

And got the message: 
```
Input files must not be processed by assembleContigs without --assemble-contigs-by option 
```

It is unclear from the manual what options the `--assemble-contigs-by` can take. Could you recommend it? Thanks!

**Comment by PoslavskySV** [comment, marked as answer]
Hi,

BD full-length protocol is fragmented, so MiXCR uses [contig assembly](https://mixcr.com/mixcr/reference/mixcr-assembleContigs/) to reconstruct the longest-possible contig sequence ([VDJRegion](https://mixcr.com/mixcr/reference/ref-gene-features/) plus L1,L2+UTR in V part in the best case). As it is not know a priori which region will be fully covered for each cell, the length of the resulting clonotype will be varying, which is not acceptable for allelic variant inference. 

That said, you can do the following:

```
mixcr analyze bd-rhapsody-mouse-tcr-full-length \
        --assemble-contigs-by VDJRegion \
        20220923-Mm023B-BCR_R1.fastq.gz \
        20220923-Mm023B-BCR_R2.fastq.gz \
       mouse_vdj/result
```

Here the option `--assemble-contigs-by VDJRegion` will drop all clones which do not cover `VDJRegion` and for those which cover a longer feature it will cut them to just `VDJRegion`.

If you don't want to re-run full `analyze` command, you can re-run only `assembleContigs` step on `.clna` input files:

```
mixcr assembleContigs --assemble-contigs-by VDJRegion input.clna output.clns
mixcr exportClones output.clns output.tsv
```

(in next releases we will improve analyze so that it will not re-run all underlying steps)
